src/state/slices/addressBookSlice/selectors.ts

```diff
@@ -0,0 +1,65 @@
+import { createSelector } from '@reduxjs/toolkit'
+import type { ChainId, ChainNamespace } from '@shapeshiftoss/caip'
+import { fromChainId } from '@shapeshiftoss/caip'
+import { matchSorter } from 'match-sorter'
+import createCachedSelector from 're-reselect'
+
+import { addressBookSlice } from './addressBookSlice'
+
+import type { ReduxState } from '@/state/reducer'
+
+export const selectAddressBookEntries = createSelector(
+  addressBookSlice.selectors.selectAddressBookEntries,
+  entries => entries,
+)
+
+export const selectAddressBookEntriesByChainNamespace = createCachedSelector(
+  addressBookSlice.selectors.selectAddressBookEntries,
+  (_state: ReduxState, chainId: ChainId) => chainId,
+  (entries, chainId) => {
+    if (!chainId) return []
+
+    const { chainNamespace } = fromChainId(chainId)
+
+    // Filter entries that match the same chain namespace
+    // This allows addresses to work across all chains of the same type
+    // (e.g., an Ethereum address works on all EVM chains)
+    return entries.filter(entry => {
+      const entryNamespace = fromChainId(entry.chainId).chainNamespace
+      return entryNamespace === chainNamespace
+    })
+  },
+)((_state: ReduxState, chainId: ChainId): ChainNamespace | string => {
+  if (!chainId) return 'empty'
+  const { chainNamespace } = fromChainId(chainId)
+  return chainNamespace
+})
+
+type SearchQueryFilter = {
+  chainId: ChainId
+  searchQuery: string
+}
+
+const selectSearchQueryFromFilter = (_state: ReduxState, filter: SearchQueryFilter) =>
+  filter.searchQuery
+
+export const selectAddressBookEntriesBySearchQuery = createCachedSelector(
+  (state: ReduxState, filter: SearchQueryFilter) =>
+    selectAddressBookEntriesByChainNamespace(state, filter.chainId),
+  selectSearchQueryFromFilter,
+  (entries, searchQuery) => {
+    if (!searchQuery) return entries
+
+    const matchedEntries = matchSorter(entries, searchQuery, {
+      keys: [
+        { key: 'name', threshold: matchSorter.rankings.MATCHES },
```

Comment:
q: wondering if we want to block ability from adding same label over two addies for paranoia purposes or am I gnoming too much?

Reply:
Askchually think you're absolutely right! Totally fine to label two addresses the same, as a user I'm a big boi I should be able to make bad decisions if I want to.